### PR TITLE
Mechanism added to avoid main menu null item actions

### PR DESCRIPTION
**Pull request by adamjryan**
Update to the main menu / sub menus to disable the submit
button when there is no item selected to prevent processing
a null object.

### CODE REVIEW

**Comment by michaelzangl** [comment]
Looks good so far. I will test it (and probably merge it) this weekend. Thanks ;-)

**Comment by andreas-eberle** [comment]
@michaelzangl: Any progress here?

**Comment by adamjryan** [comment]
Let me know if there are any changes needed for this, so it can get sorted.

File: jsettlers.graphics/src/jsettlers/graphics/utils/UIList.java [hunk 1]
```diff
@@ -49,19 +49,18 @@
...
 		this.itemheight = itemheight;
-		if (items.size() > 0) {
-			activeItem = items.get(0);
-		}
 	}
 
 	public void setItems(List<? extends T> list) {
 		synchronized (itemsMutex) {
 			this.items = list;
+			if (items.size() > 0) {
```
Comment (michaelzangl) [inline jsettlers.graphics/src/jsettlers/graphics/utils/UIList.java]: This is not the way it was intended. Lists are often updated by adding items. If we reset the active item on every list update, we reset the user selection.

Comment (adamjryan) [inline jsettlers.graphics/src/jsettlers/graphics/utils/UIList.java]: If a new set of items is set for the UIList then the previous active item would be invalid. In fact when the new list size is 0 the active item should be set to null. There should also really be a check on the setActiveItem method as well to check it's in the list.
Additions would be made on the List object (that's passed in) and these will be reflected in the UI without changing the user's selection.

Comment (adamjryan) [inline jsettlers.graphics/src/jsettlers/graphics/utils/UIList.java]: I think maybe you meant that you want the previous selection to be kept when going back to a menu. I've updated it now so the active item for each menu is not reset when clicking between menus.

**Comment by michaelzangl** [comment]
Merged in controls-improve-menu.

**Pull request closed without merging**